stacks-network org README

wileyj opened the pull request:
re-opening this for comment/review from reverted #3 . 

This will be the display page for the `stacks-network` github org page, with links directing people to resources relevant to what we do. 

This commit is a copy/paste from a skiff doc; it still needs a bit of formatting and updates (images etc). 
This PR is to start the process and get some discussion going on what it should look like when it is merged. 


comments/suggestion for changes are welcome here, or you can PR any changes to the `chore/add-readme` branch


@muneeb-ali tagging you since you mentioned this in a call

wileyj committed:
initial copy/paste

Commit:
Update readme and add Stacks Core Eng docs

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/stacks-network/.github?pullRequest=5) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/stacks-network/.github?pullRequest=5) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: wileyj<br/>:x: Will Corcoran<br/><hr/>**Will Corcoran** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/stacks-network/.github?pullRequest=5) it.</sub>

Commit:
Update readme links

wileyj commented:
https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/customizing-your-organizations-profile

the folder structure needs to change to match what is here. i.e. put the files that need to be displayed at the org level under the `./profile` folder

wileyj commented on clarity-vm-tsd.md:
```diff
@@ -0,0 +1,89 @@
+# Clarity VM | Technical Specification Document
+
+# Executive Summary:
```
These H1 headers below line 1 should all be at least H2 `##` or H3 if a subsection

wileyj commented on sbtc-signer-prd.md:
```diff
@@ -0,0 +1,100 @@
+
+The system should automatically initiate a manual review process for specific transactions based on predefined conditions to ensure compliance and prevent suspicious activities.
+
+**Requirements:**
```
would this serve better as a checkbox, or just a list? a table doesn't seem right here

wileyj commented on stacks-core-eng-home.md:
```diff
@@ -0,0 +1,100 @@
+
+>OKRs forthcoming
+
+## 3Q23 OKRs: sBTC
```
This may work better as a table vs a list

wileyj commented on nakamoto-release-tsd.md:
With a file this large, it would be helpful to:
1. break it apart into multiple files
2. add an index to the markdown so it's easier to navigate

wileyj commented:
I'm also not entirely sold on the idea of having *all* of the documentation in this repo. would it make more sense to have only the README, and move supporting docs to another repo, like https://github.com/stacks-network/stacks ?

wileyj committed:
Cleaning up

- remove sbtc docs (to be added to `stacks` repo as a pr)
- revert template README in the root
- update profile/readme with image
- move banner image to profile dir
- add standalone images
- Bump the license terms for year and org
- update code of conduct to match what is used for stacks-blockchain repo

wileyj committed:
update formatting

adding the stacks logo as the top item
move the banner to the bottom

wileyj commented:
@marthaadrienne i'm going to move the documents you've commented on to the https://github.com/stacks-network/stacks repo with your suggestions in mind. 

they've been removed from this PR, but would appreciate you taking a second look at https://github.com/stacks-network/.github/blob/chore/add-readme/profile/README.md for the choices i made there

@will-at-stacks thoughts on teh formatting changes to the readme?

wileyj commented on stacks-core-eng-home.md:
```diff
@@ -0,0 +1,100 @@
+    - **Measure:** Percentage of open questions answered by docs.
+    - **Target:** 90%
+- **Key Result 3:** The Clarity smart contracts of the sBTC design are deployed
+    - **Measure:** Percentage of defined contracts deployed
```
all periods removed. i also think i'll be converting this to a table in the PR to the stacks repo

wileyj commented on testing-and-hardening-tsd.md:
```diff
@@ -0,0 +1,66 @@
+
+# Solution:
+
+>👉 *How does the* Testing & Hardening *initiative provide a solution to the identified problem?*
```
done. i've also standardized the other "questions" in the doc so they're all italicized

wileyj commented on clarity-vm-tsd.md:
```diff
@@ -0,0 +1,89 @@
+
+The Clarity VM in the stacks-node faces inefficiencies in transaction processing, limiting the number of transactions that can be included in a block. This hampers user experience and transaction capacity. An improved VM is essential to address these limitations.
+
+With an improved Clarity VM, transactions can be processed much faster. This enhancement will allow us to increase the number of transactions that can fit into each block and improve the overall user experience significantly.
```
a good point - i think i'll make this change for the PR:
```
will allow us to increase the number of transactions
```
->
```
will enable an increase in the number of transactions
```

wileyj commented on testing-and-hardening-tsd.md:
```diff
@@ -0,0 +1,66 @@
+
+The Testing & Hardening initiative proposes a multi-pronged solution: implement effective black-box testing to detect bugs and improve test execution for pull requests.
+
+First, it’s important to note that these bugs weren’t missed for lack of trying— some bugs are unavoidable, but these bugs are mostly not impossibly complex, and so its reasonable to think that some kind of testing would have caught them. For these kinds of bugs, black-box testing is very well suited to uncovering them. So, the first prong of this work is to implement sufficient black-box testing ...
```
or the verbose `it is`, i shall PR that change

marthaadrienne reviewed (approved):
Reviewed with no changes. Thank you.

will-at-stacks commented:
Hey @wileyj just making my way through these comments.  All of the suggested format revisions look good to me.  Thanks!


> @marthaadrienne i'm going to move the documents you've commented on to the https://github.com/stacks-network/stacks repo with your suggestions in mind.
> 
> they've been removed from this PR, but would appreciate you taking a second look at https://github.com/stacks-network/.github/blob/chore/add-readme/profile/README.md for the choices i made there
> 
> @will-at-stacks thoughts on teh formatting changes to the readme?

will-at-stacks commented:
@wileyj It looks like the primary Stacks Core Eng is now linking to a Skiff.  Is that correct?  Goal is to get this information all on GH.

wileyj commented:
> @wileyj It looks like the primary Stacks Core Eng is now linking to a Skiff. Is that correct? Goal is to get this information all on GH.

what should it be changed to?

will-at-stacks commented:
it should be pointing to stacks-core-eng-home.md

>

wileyj commented:
> 

i see. that file is part of another, much larger PR that is in draft status https://github.com/stacks-network/stacks/pull/485

would it make sense to remove that link from the readme until it's ready?

kenrogers commented on nakamoto-release-tsd.md:
```diff
@@ -0,0 +1,973 @@
+
+## In summary, The Nakamoto Release:
+
+- Replaces competitive mining (ratified in SIP-001 and SIP-00 with a cooperative system between miners and PoX stackers through Byzantine fault-tolerant (BFT) agreement that produces a linearized stream of confirmed transactions with periodic chainstate recording to Bitcoin blockchain.
```
Number missing here in second SIP

wileyj replied:
https://github.com/stacks-network/stacks/pull/485

kenrogers reviewed (approved):
This is looking good to me, structure makes sense, just some minor suggestions on some formatting and grammatical mistakes.

wileyj commented on clarity-vm-tsd.md:
```diff
@@ -0,0 +1,89 @@
+
+The Clarity VM (Virtual Machine) is like the brain of the Stacks blockchain. It's responsible for understanding and executing special programs written inthe smart contracting language, Clarity. These programs are like sets of instructions that tell the blockchain how to perform various tasks and transactions securely.
+
+The current Clarity VM can sometimes be slow in processing these instructions, which can cause delays in transactions. To make the blockchain faster and more efficient, the Clarity VM Workstream is working on a solution. They plan to use a special technology called WASM (WebAssembly), which is known for its speed and safety, to make the Clarity VM work much faster.
```
https://github.com/stacks-network/stacks/pull/485

wileyj commented on sbtc-prd.md:
```diff
@@ -0,0 +1,124 @@
+
+sBTC is a protocol enabling secure movement of Bitcoin in/out of the Stacks blockchain for fully decentralized, scalable Bitcoin applications. It implements the SIP10 token standard and is managed by a decentralized group of signers, known as Stackers, on the Bitcoin main chain.
+
+
```
https://github.com/stacks-network/stacks/pull/485

wileyj commented on sbtc-tsd.md:
```diff
@@ -0,0 +1,60 @@
+
+>👉 *What can you learn from exploring the Product Requirements Document for sBTC? Discover its purpose, target audience, benefits, problem-solving capabilities, and implementation details.*
+
+Introducing a revolutionary protocol on the Stacks chain, enabling secure movement of Bitcoin and fully decentralized applications for the Bitcoin currency. Discover how this non-custodial digital asset, sBTC, facilitates seamless transactions and unlocks new possibilities in DeFi while preserving the core principles of decentralization and self-sovereignty. Explore how its 1:1 BTC backing and de ...
```
https://github.com/stacks-network/stacks/pull/485

wileyj commented on clarity-vm-tsd.md:
```diff
@@ -0,0 +1,89 @@
+
+</br>
+
+The plan to speed up the Clarity VM is to compile Clarity contracts into [WebAssembly (Wasm)](https://webassembly.org/), and then integrate a Wasm runtime into the Clarity VM. Wasm has proven to be an effective tool for fast, safe, and portable code execution, and is being used in several other blockchain projects. Initial back-of-the-envelope testing shows that a Wasm runtime has the potential t ...
```
https://github.com/stacks-network/stacks/pull/485

will-at-stacks commented:
@wileyj thanks for your work on this!  question, is it still going to be held up by me being flagged as not having signed the CLA?  The error message is below... it looks like I was not signed into the correct email address via terminal when I pushed this commit.  I've since logged in and signed the CLA - but the error does not want to clear... any thoughts?

> Will Corcoran seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/stacks-network/.github?pullRequest=5) it.

wileyj commented:
> @wileyj thanks for your work on this! question, is it still going to be held up by me being flagged as not having signed the CLA? The error message is below... it looks like I was not signed into the correct email address via terminal when I pushed this commit. I've since logged in and signed the CLA - but the error does not want to clear... any thoughts?
> 
> > Will Corcoran seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> > You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/stacks-network/.github?pullRequest=5) it.

outside of the question i posed  in https://github.com/stacks-network/.github/pull/5#issuecomment-1652586387, the CLA won't be an issue for this PR

wileyj committed:
Updating link once stacks-network/stacks #485 is merged

wileyj committed:
align image to the left

wileyj commented:
now that https://github.com/stacks-network/stacks/pull/485  is merged, merging this one
